src/main/java/com/ridemart/entity/Advertisement.java

```diff
@@ -0,0 +1,59 @@
+package com.ridemart.entity;
+
+import jakarta.persistence.*;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Entity
+@Data
+@NoArgsConstructor
+public class Advertisement {
```

Comment:
Consider adding AllArgsConstructor in your Entities. Then go over the Services, where you use these Entities and see if it is not possible to use the AllArgsConstructor perhaps.